app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkChaptersCompletedActivityTest.kt

```diff
@@ -750,6 +750,116 @@ class MarkChaptersCompletedActivityTest {
     }
   }
 
+  @Test
+  fun testMarkChaptersCompletedActivity_clickMarkCompleted_activityFinishes() {
+    launch<MarkChaptersCompletedActivity>(
+      createMarkChaptersCompletedActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.mark_chapters_completed_mark_completed_text_view)).perform(click())
+      assertThat(activityTestRule.activity).isNull()
+    }
+  }
+
+  @Test
+  fun testMarkChaptersCompletedActivity_configChange_clickMarkCompleted_activityFinishes() {
+    launch<MarkChaptersCompletedActivity>(
+      createMarkChaptersCompletedActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.mark_chapters_completed_mark_completed_text_view)).perform(click())
+      assertThat(activityTestRule.activity).isNull()
+    }
+  }
+
+  @Test
+  fun testMarkChaptersCompletedActivity_selectMultipleChapters_clickMarkCompleted_reopenActivity_completedChaptersAreChecked() { // ktlint-disable max-line-length
+    launch<MarkChaptersCompletedActivity>(
+      createMarkChaptersCompletedActivityIntent(internalProfileId)
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 1)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 1)
+      scrollToPosition(position = 4)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 4)
+      scrollToPosition(position = 9)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 9)
+      scrollToPosition(position = 10)
+      performItemCheckOnRecyclerViewItemAtPosition(itemPosition = 10)
+      onView(withId(R.id.mark_chapters_completed_mark_completed_text_view)).perform(click())
+    }
+    launch<MarkChaptersCompletedActivity>(
```

Comment:
@anandwana001 Is this a good way to write tests? I meaning launching it twice in same test.

Reply:
not its not, make separate test cases.

Reply:
Actually what I need to test is here is that when the user clicks the `MARK COMPLETED` button, the activity finishes and then when the user reopens the activity the completed items are already marked. Could you please suggest me some better ways to reopen an activity?

Reply:
check this in a way
after click - as it is opening new activity - check `intended`

then for the next activity, try to pre-fill data and open the activity and test it, so this way we can have 2 separate test testing these individually, not dependent. 

Also, such test will fail on robolectric, so test one activity at a time.

Reply:
Oh okay. In that case, I don't think we need this test as I have already written tests that check for the individual functionalities. Removing this test now. Thanks.